core/instance/container.go

```diff
@@ -14,12 +14,14 @@ type ContainerType string
 const (
 	NONE ContainerType = "none"
 	LXD  ContainerType = "lxd"
+	LXC  ContainerType = "lxc"
```

Comment:
So this is wrong. We don't support `lxc` directly, we use `lxd` to drive `lxc`. Adding this would break a lot of things in nested containers. The output from the executed command should be parsed from `lxc` to `lxd`.

You will have to be careful that you don't also break the `to` placement. We do not want `--to lxc:0`. The comments around the placement are also incorrect (lxc instead of lxd):

https://github.com/juju/juju/blob/be740127699aefc07321a9504ac79d00f856a9bf/core/bundle/changes/changes.go#L447

Reply:
Also `systemd-detect-virt` reports `qemu` rather than `kvm` on older releases under lxd vms[1]. If this is a possible use-case, we may want to handle that.

[1] https://github.com/systemd/systemd/issues/22709